arch/sim/src/sim/up_internal.h

```diff
@@ -232,6 +232,8 @@ void  host_free_shmem(void *mem);
 uint64_t host_gettime(bool rtc);
 void host_sleep(uint64_t nsec);
 void host_sleepuntil(uint64_t nsec);
+int host_setup_timer(void);
```

Comment:
How about we let host_setup_timer return(or add a pointer) the irq number?
```
int host_setup_timer(int *irq);
```
so we don't need host_get_alarm_signal.

Reply:
I will update this, thanks

Reply:
The sequence isn't important here, it's fine to call host_settimer, irq_attach and irq_enable.